src/views/ApartmentApp/components/StudentApplication/components/HallChoiceList/components/HallListItem/index.js

```diff
@@ -102,8 +68,11 @@ const HallListItem = ({
               <InputLabel>Rank</InputLabel>
               <Select
                 disabled={disabled}
-                value={hallRank}
-                onChange={handleRankInputChange}
+                value={HallRank}
+                onChange={(event) =>
+                  event.target.value !== null &&
+                  onHallInputChange?.(String(event.target.value), HallName, index)
+                }
```

Comment:
Were you getting instances where `event.target.value` was null?

Reply:
I think so? 
I remember there are weird edge-cases where things do strange behaviors as the page is loading,...

Though now that I think about it, the errors with callbacks getting triggered on declaration was something that happened when using class components, unless you put some extra stuff in place, 
and I think the way the functional components handle things prevents those unwanted behaviors, so I might be no longer an issue

Reply:
@NerdyGriffin  Note to self: test this out later